PhIPSeq_external/Analysis_allergens_and_IEDB/predict_age.py

```diff
@@ -0,0 +1,121 @@
+import os
+import string
+import seaborn as sns
+import numpy as np
+import pandas as pd
+from matplotlib import pyplot as plt
+from scipy.stats import fisher_exact
+from sklearn.metrics import r2_score
+from statsmodels.stats.multitest import fdrcorrection
+
+from PhIPSeq_external.Analysis_allergens_and_IEDB.config import base_path, out_path
+from PhIPSeq_external.Analysis_allergens_and_IEDB.prediction_helpers import get_oligos, configure_x_y, \
+    tune_hyper_parameters, predict_with_cross_validation, plot_shap, read_metadata
+
+
+def compute_age_correlations(x, y, overwrite=True):
+    csv_out_path = os.path.join(out_path, 'age_correlations_data.csv')
+    if overwrite or not os.path.exists(csv_out_path):
+        os.makedirs(out_path, exist_ok=True)
+        bottom_quintile = y.quantile(0.2)
+        top_quintile = y.quantile(0.8)
+        bottom_label = f"Percent of youngest quintile (<{int(bottom_quintile)}\nin whom a peptide is significantly bound"
+        top_label = f"Percent of oldest quintile (>{int(top_quintile)}\nin whom a peptide is significantly bound"
+        age_group = y.apply(
+            lambda v: bottom_label if v < bottom_quintile else (top_label if v > top_quintile else np.nan))
+        oligos = x.drop(columns=['gender']).gt(1)
+        oligos['age_group'] = age_group
+        oligos.dropna(inplace=True)
+
+        # Perform Fisher's exact test to identify significant oligos
+        summed_outcome = age_group.value_counts()
+        summed_df = oligos.groupby('age_group').sum().T
+        summed_df.rename(columns={k: f"{k}_1" for k in summed_df.columns}, inplace=True)
+        for col in summed_outcome.index:
+            summed_df[f"{col}_0"] = summed_outcome.loc[col] - summed_df[f"{col}_1"]
+        summed_df['oddsratio'] = summed_df.apply(
+            lambda row: fisher_exact([[row[bottom_label + '_0'], row[bottom_label + '_1']],
+                                      [row[top_label + '_0'], row[top_label + '_1']]])[0], axis=1)
+        summed_df['p-value'] = summed_df.apply(
+            lambda row: fisher_exact([[row[bottom_label + '_0'], row[bottom_label + '_1']],
+                                      [row[top_label + '_0'], row[top_label + '_1']]])[1], axis=1)
+        summed_df['passed_fdr'], summed_df['fdr_corrected_p_value'] = fdrcorrection(summed_df['p-value'])
+
+        oligos = oligos.groupby('age_group').mean().T * 100.0
+        oligos = oligos.merge(summed_df[['oddsratio', 'p-value', 'passed_fdr', 'fdr_corrected_p_value']],
+                              left_index=True, right_index=True, how='left')
+        df_info = pd.read_csv(os.path.join(base_path, "library_contents.csv"), index_col=0, low_memory=False)[
+            ['nuc_seq', 'full name']]
+        oligos = oligos.merge(df_info, left_index=True, right_index=True, how='left')
+        oligos.to_csv(csv_out_path)
+    return pd.read_csv(csv_out_path, index_col=0)
```

Comment:
why not return what we have, and only read if did not compute?
I.e.:
             return oligos
    return pd.read_csv(csv_out_path, index_col=0)